internal/resources/private_endpoint_service.go

```diff
@@ -0,0 +1,348 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &PrivateEndpointService{}
+	_ resource.ResourceWithConfigure   = &PrivateEndpointService{}
+	_ resource.ResourceWithImportState = &PrivateEndpointService{}
+)
+
+const (
+	errorMessageWhileEnablingPrivateEndpointService = "There is an error while enabling private endpoint service. Please check in Capella to see if there are any hanging resources\" +\n\t\" have been created, unexpected error: "
+)
+
+type PrivateEndpointService struct {
+	*providerschema.Data
+}
+
+func NewPrivateEndpointService() resource.Resource {
+	return &PrivateEndpointService{}
+}
+
+func (p *PrivateEndpointService) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_private_endpoint_service"
+}
+
+func (p *PrivateEndpointService) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"organization_id": stringAttribute([]string{required}),
```

Comment:
Is there will any impact if we add requireReplace here

Reply:
yes.  if org id changes, then terraform will destroy/create the resource.  

as we discussed, it's a better user experience.  without `requireReplace` plan will indicate an update, but apply will fail.

further, if org/project/cluster changes, it makes no sense to call update function.

Reply:
addressed in 908d1ad